### PR TITLE
ARROW-7852: [Python] 0.16.0 wheels not compatible with older numpy

### PR DESCRIPTION
**Pull request by kszucs**

### CODE REVIEW

**Comment by kszucs** [comment]
@github-actions crossbow submit -g wheel

**Comment by kszucs** [comment]
@github-actions crossbow submit -g wheel

**Comment by kszucs** [comment]
@github-actions crossbow submit -g wheel

File: python/requirements-wheel-build.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Keras-Preprocessing
+cython>=0.29.11
+numpy==1.14.5; python_version < "3.8"
```
Comment (jorisvandenbossche) [inline python/requirements-wheel-build.txt]: In pandas we build against numpy==1.13.3 for python 3.6 (and 1.14.5 for python 3.7), do we have a documented minimum required numpy version for pyarrow?

Comment (kszucs) [reply python/requirements-wheel-build.txt]: 1.14 is pinned in setup.py and originates from https://issues.apache.org/jira/browse/ARROW-3141

I don't think we have a documentation for this.

File: python/requirements-wheel-test.txt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+cffi
+cython
+hypothesis==5.0; python_version <= "3.5.2"
```
Comment (pitrou) [inline python/requirements-wheel-test.txt]: Why "3.5.2" exactly? Did something happen in 3.5.3?

File: python/requirements-test.txt [hunk 3]
```diff
@@ -1,5 +1,4 @@
 cffi
-cython
```
Comment (pitrou) [inline python/requirements-test.txt]: Why do you remove this? There's a Cython test.

Comment (kszucs) [reply python/requirements-test.txt]: Becuase we usually pass bot requirements files and defining a package twice makes pip unhappy.

File: python/requirements-wheel-test.txt (hunk 2 above)
Comment (kszucs) [reply python/requirements-wheel-test.txt]: Yes, the python typing incompatibilities happened at that version.

**Comment by kszucs** [comment]
The 3.5 failures depend on #6762

**Review by pitrou — APPROVED** [review]
+1 from me.

**Pull request closed without merging**